Server Framework

Dejaku51 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
This PR makes it so servers (currently RnD and tcomms) generate heat and work with efficiency depending on their temperature

Also adds cooling to kilo, rad and meta tcomms

Makes RnD servers generate less heat and use less electricity with better parts

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
No more 4 part tcomms in random room without proper setup, cooling tcomms actually has sense now
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
-->

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
1. Test if tcomms work roundstart (they do)
2. turn off freezer in tcomms
3. say different things
4. as the temperature in tcomms rises, efficiency of machines lowers and messages get sent with lag depending on efficiency of the machines handling the message
<details>
<summary>Screenshots&Videos</summary>
Lower efficiency because the server is hot

![image](https://github.com/BeeStation/BeeStation-Hornet/assets/40302913/fb02b691-44d1-4a46-acc1-af8954b902aa)

![image](https://github.com/BeeStation/BeeStation-Hornet/assets/40302913/706044f3-dbf0-4877-acda-491293e78281)



Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog
:cl:
refactor: Tcomms and RnD servers now generate heat and work with varying efficiency based on the temperature it is
tweak: Added cooling to tcomms in metastation
tweak: Added cooling to tcomms in kilostation
tweak: Added cooling to tcomms in radstation
tweak: RnD servers now use less power and generate less heat if they got better parts
code: changed typepaths of tcomms machines from /obj/machinery/telecomms to /obj/machinery/server/telecomms
code: Added connect_roundstart to atmos connector that will make it automatically connect to any portable atmos machine (ex. canister) on the same tile when spawned
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Dejaku51 committed:
enjoy reviewing it

Dejaku51 committed:
lets try use atmos code to handle that? (experiment)

Dejaku51 committed:
leftover

Dejaku51 committed:
guh

Dejaku51 committed:
nah

Dejaku51 committed:
Merge branch 'master' of https://github.com/BeeStation/BeeStation-Hornet into servers

Dejaku51 committed:
guh

Dejaku51 committed:
Merge branch 'master' of https://github.com/BeeStation/BeeStation-Hornet into servers

Dejaku51 committed:
GAHHH NEW FOOD

Dejaku51 committed:
much better

Co-authored-by: Penwin <53474257+Penwin0@users.noreply.github.com>

Dejaku51 committed:
Update server temperature calculation and heat transfer

Dejaku51 committed:
delete debug stuff

Dejaku51 committed:
RnD servers

Dejaku51 committed:
reset map changes

Dejaku51 committed:
fix server paths on Kilo Meta Rad

Dejaku51 committed:
cooling in tcomms once again

Dejaku51 committed:
that hole shouldn't be there

Dejaku51 committed:
tcomms throttling

EvilDragonfiend commented on code/__DEFINES/stat.dm:
```diff
@@ -12,10 +12,12 @@
-#define EMPED	(1<<3)		// temporary broken by EMP pulse
+#define BROKEN		(1<<0)
+#define NOPOWER		(1<<1)
+#define MAINT		(1<<2)		// under maintaince
```
since you're here fix the typo lol

Dejaku51 replied:
fixed

Dejaku51 committed:
Fix typo in stat.dm

Dejaku51 committed:
Updates!

Co-authored-by: PestoVerde322 <75247747+PestoVerde322@users.noreply.github.com>
Co-authored-by: BriggsIDP <107176252+BriggsIDP@users.noreply.github.com>

Miliviu reviewed (changes requested):
Hi! I'm not a maintainer but I've checked the code and LGTM!

But I have a concern! I was testing the RnD Server Power consumption (And it works)
<details>
Before cool parts:

![image](https://github.com/BeeStation/BeeStation-Hornet/assets/45698448/4594bf98-fd90-42dc-9242-c61196439cb4)

After cool parts:

![image](https://github.com/BeeStation/BeeStation-Hornet/assets/45698448/8cf4466a-7b30-4361-9e9b-76a9eb907847)

</details>

But I couldn't upgrade the Servers, the only way it was to "destroy" the server and build it with new parts

<details>
Here, you can see I'm trying to deconstruct (or just open the latch with the screwdriver a turned off server and I can't 

![image](https://github.com/BeeStation/BeeStation-Hornet/assets/45698448/3bcc80db-c30e-4582-99c6-313df4b27d29)
</details>

I'm not sure if this intended, if it is then I'd not see any issues with the PR :)

Dejaku51 committed:
you can now upgrade tcomm servers

Dejaku51 committed:
remove unwanted changes to runtime station

Miliviu reviewed (approved):
Yup! That should do it, LGTM!

itsmeow reviewed (changes requested):
Please add an UpdatePaths script for the new servers. (in `tools/UpdatePaths/Scripts`)

Dejaku51 committed:
did i do it right?

Dejaku51 committed:
Merge branch 'master' into servers

Dejaku51 committed:
forgot to remove the bad code

Dejaku51 committed:
Merge branch 'servers' of https://github.com/Dejaku51/BeeStation-Hornet into servers

Miliviu committed:
Modify Wired Glass output, if you get mats unbuilding one, you shouldn't get 5 with the same mats! (#10446)

EvilDragonfiend committed:
Fixes an abstract type of an object throwing runtime on smooth_icon() (#10403)

* no runtime

* crash doc

Tsar-Salat committed:
More spawners guh (#10402)

Tsar-Salat committed:
Ports post-basemat fixes to cyborg stacks (#10380)

* https://github.com/tgstation/tgstation/pull/54935

* https://github.com/tgstation/tgstation/pull/55858

Rukofamicom committed:
Random events no longer indicate whether they were triggered by the random event system (#10379)

* Removes confusion and salt

* forgot to push the second line

ss13-beebot committed:
Automatic changelog generation for PR #10446 [ci skip]

ss13-beebot committed:
Automatic changelog generation for PR #10380 [ci skip]

ss13-beebot committed:
Automatic changelog generation for PR #10379 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Miliviu committed:
Readds a check removed by a bugfix! (#10453)

HowToLoLu committed:
Fixes The Atmos Bug (#10451)

* Yes, fixing *THE* atmos bug, the one that's been happening for a year now.

* Why why and *why*

* No lag from you

* Even more warning to the coder

* Just delete these actually

ss13-beebot committed:
Automatic changelog generation for PR #10451 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

HowToLoLu committed:
Unit Test for Smoothing (and fixes for some uncaught issues) (#10066)

* Some Icon State Fixes

* Bump

* Unit Test: Computer Smoothing

* NONE not null

* That should be the last of them

* Revert "Unit Test: Computer Smoothing"

This reverts commit 7d0d4ca45c2466ec1cc3a6521ebea32407bd656c.

* Clean that up there.

* Some fixes for various smoothing atoms

* The test (There will be a ton of failures related to /turf/closed/mineral)

* yolo test

* whoopsies

* should've actually initialized the atoms from the get-go

* Allocate is better. Mostly

* Runtime fix

* Missing icon stuff

* Group order matters I guess

* Yes, this PR is still alive, no the errors aren't going away gaaaah

* How did I not notice this?

* Well I think it'll work now.

* Left that in there by accident

* Final touches to this file...

* Hopefully a fix for this null

* Xenoartifact console runtime...

* Undoes some stuff done to ashplanet flooring

* Different solution more correct result

* Oh hey we got our own icon for that now.

* Falsewall icon fixes

* Rename - Honestly should've done this from the beginning

* Hopefully fixes these (cursed) effects

* Whoopsie!

* Looking at it now; I get a better understanding of why

* why smoothing is done in certain ways.

* Apparently there is no lag concern and it was all in my head

* Effectively a revert on this file

* Not quite there yet...

* Revert "Not quite there yet..."

This reverts commit b20210b1b03dc2b10e7f898d7541628499333c41.

* No need for this comment

* this might prevent the quadrupled error output

* *NOW* it's fixed; apparently I was still doing wierd stuff before.

* No, this is objects

* WHAT ARE YOU DOING OVER THERE; HOW ARE YOU OVER THERE?

* Comments, proper order, and now smoothing right.

* And while we're at it these have been bugging me for a while

* So that we don't error

* Okay that should be it

* Ash rock fixed, unit test ftw

* Alien weeds should be better codewise now

* Wrong define name to check

PestoVerde322 committed:
Crates and locker resprite and overhaul (#10136)

* Crate time

* icon dir fix

* a

* more icon thingies

* cratery and lockery

i need to finish lockers and bring a new mulebot to the table

* icons done, mostly code todo

* major update

all storage is properly separated

* Crates and Lockers complete!

* return those to default

* fixed crate climbing

* runtimes ree

* mmmmm done

* fixes crates and lockers

no more runtimes

* finished the wrapping and fixed some stuff

* how did i miss beestation.dme?

* unticking funny

* revert runtimestation.dmm

---------

Co-authored-by: PigeonVerde322 <pesto.verde322@gmai.com>

ss13-beebot committed:
Automatic changelog generation for PR #10066 [ci skip]

Tsar-Salat committed:
Ports refactor of electrocute_act, switches out args for flags (#10300)

* https://github.com/tgstation/tgstation/pull/45291

* https://github.com/tgstation/tgstation/pull/45435

* https://github.com/tgstation/tgstation/pull/46564

* https://github.com/tgstation/tgstation/pull/47666

ss13-beebot committed:
Automatic changelog generation for PR #10136 [ci skip]

ss13-beebot committed:
Automatic changelog generation for PR #10300 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Rukofamicom committed:
Fixes trashbag with quick equip belt hotkey (#10133)

* Fixes trashbag with hotkeys

* Update human.dm

ss13-beebot committed:
Automatic changelog generation for PR #10133 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

PowerfulBacon committed:
MC Diagnostic Mode (#10059)

* MC Diagnostic Mode

* Update master.dm

* Changes the percentage to a millisecond value instead

* Update master.dm

* Update mob_stat.dm

ss13-beebot committed:
Automatic changelog generation for PR #10059 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ToasterBan committed:
Fixes the Chicken Nugget Box (#10461)

Tsar-Salat committed:
removes overthrow gamemode (#10434)

Nakalon committed:
White Bandana is real (#10425)

Adds white bandana

ss13-beebot committed:
Automatic changelog generation for PR #10461 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog generation for PR #10425 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

PestoVerde322 committed:
Fixes Carpets Runtime when they get burned by atmospherics (#10339)

Co-authored-by: PigeonVerde322 <pesto.verde322@gmai.com>

ss13-beebot committed:
Automatic changelog generation for PR #10339 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ToasterBan committed:
Wielded Mjollnir is visible again (#10471)

ss13-beebot committed:
Automatic changelog generation for PR #10471 [ci skip]

ClownMoff committed:
Adds a Plushie Delivery Beacon to the game. (#10398)

* :·3

* fix

* among us sound not real :(

* among us plushie legs fix

* Update plushes.dmi

* Update plushes_lefthand.dmi

* :mouse:

* Update plushes.dm

* gondola

* Update plushes.dmi

* Gondola

* Update plushes.dm

* Update plushes.dm

* Update plushes.dm

* Update plushes.dm

* fix

* YES!

* Final touches

* Update plushes.dm

* Update games.dm

* Trait change

* Update positive_quirk.dm

* Update positive_quirk.dm

* Update plushes.dmi

* Update plushes.dm

* Update plushes.dmi

* Update code/game/objects/items/plushes.dm

Co-authored-by: Dejaku51 <40302913+Dejaku51@users.noreply.github.com>

* Update code/game/objects/items/plushes.dm

Co-authored-by: Lucy <lucy@absolucy.moe>

* Update code/game/objects/items/plushes.dm

Co-authored-by: Lucy <lucy@absolucy.moe>

* Summary (Required)

* Update code/datums/traits/positive_quirk.dm

Co-authored-by: itsmeow <itsmeow@itsmeow.dev>

* removes nar'sie and rav'tar plushies

* Update games.dm

---------

Co-authored-by: RKz <falldownftw@gmail.com>
Co-authored-by: Dejaku51 <40302913+Dejaku51@users.noreply.github.com>
Co-authored-by: Lucy <lucy@absolucy.moe>
Co-authored-by: itsmeow <itsmeow@itsmeow.dev>

ss13-beebot committed:
Automatic changelog generation for PR #10398 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Absolucy committed:
Fix monkeys ripping apart DNA scanners from the inside (#10356)

* Fix monkeys ripping apart DNA scanners from the inside

* Check isturf

Tsar-Salat committed:
Restricts changing cable color to cyborgs (#10458)

* this should do

* test code begone

* as requested

ss13-beebot committed:
Automatic changelog generation for PR #10356 [ci skip]

ss13-beebot committed:
Automatic changelog generation for PR #10458 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

itsmeow committed:
Dynamic: No Lowpop Wizard - Lategame fixes (#10470)

* Dynamic: Fix Lowpop Wizard Spam

* more fixes

ss13-beebot committed:
Automatic changelog generation for PR #10470 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Haliris committed:
Ports Genpop from NSV (#10166)

* Genpop

Ports genpop from NSV

* Add construction and deconstruction support

* Updates UI icons, implements space law modifiers

* Add the ability to add descriptions to crimes, presets will now copy their tooltip on the record of incarceration.

* Fixes radio messages broadcasts

* Adds support for workshop holodeck in prison, adds alarm grilles and secured floors, also renames prisoner lockers

* Boxstation V1, two workshop templates

* Adds prison_scanner, tweaks to brig and workshop templates, makes the crafting menu see hologram items

Adds the prison scanner, which allows prisoner to scan donuts and bots they'll have made to get sentence reductions.

I had to modify crafting.dm, so if it causes any problems, this is the commit to look at.

* Tries to adress checks failing

Funny airlock over wall, and weird fucking vent in the detective office for some reasons?

* Finally found where tgui-prettier was

* Prisoners can't drag each other through the turnstile

* Workshop randomly loads a program at roundstart

* Code improvements, can't get that search bar to work in the genpop.js

* Updates bot workshop

* fixes bad ids for prison lockdown

* Adds emergency shutdown button for the workshop, adds workshop UI file,  cleans up code in various files, tweaks to workshop templates.

* Removes useless check in workshop.dm

* Fixes double quotation marks in a tutorial paper

* Fixes check in workshop emergency shutdown

* Disables bot workshop, adds plushes sweatshop instead. Add plushes crafting method.

* Adds welder step to AI deconstruction. Prevents buckled prisoners from being allowed through the turnstile.

* Makes superficial map adjustments based on feedback. Fixes workshop runtime because of holo_item being used instead of spawned, adds visible error message when someone tries to pass through the turnstile with a ridden vehicle.

* Adresses feedback

Tweaks Brig according to feedback (bad door in solitary cell, interrogation room simplified, labor shuttle dock corrected, other minor stuff).
Fixed workshop UI not checking for any access, added proper construction costs to turnstile and deconstruction hints.
Added warning and block to secoffs trying to extend the sentences of prisoner already released, not ideal, find another solution later.

* Commit 1

* Executes prettification

* Update genpop.dm

* Re-adds the tooltips

* WIP Config

All fucked up and shit

* bad line

* Finally got it working

* Adds first draft of Delta

* Bunch of hotfixes for delta

* Removes bad lattices

* I hate lattices

* Adds meta

* Adds corg, other map tweaks

* Carrots exist again, thanks newfood or something I guess

* swaps rustg for file2text to check for runtime in check

* Fland

* Adds Fland

* Adds kilo

* Adds Rad

* Fucking lattice and grille

* Makes the prisoner interface a buildable wallframe instead

* Fixes the regex

* Removes the bot workshop and associated code.

* Adresses various requests

Fixes most requests in grille and prison_floor.
Removed redundancies in workshop.dm.
Added bitfield for obj_scanned flag
use() in toys.dm for plushes making

* Adresses mapping oversight.

Removed leftover rooms in delta and kilo
Changes r_walls distribution in kilo brig
fixes escape pod on delta

* Bad linters

* Excludes dehydrated carps from plushes making

* Adresses various requests in genpop.dm

* Woopsie fucky doozy

* Finishes adressing requests on genpop.dm, ignore the else without if clause

* Bad check in prison_floor

* Fix punctuation and qdell null

* Finishes adressing changes

* Fixes linters

* Improves crafting menu check for holo items, fixes yet more newfood paths

* Adresses plating and prison_floor requests

* Fixes powernet on box and bad maint

* Fixes missing labor shuttle dock on Corg

* Does this fix checks? Also adresses oversight in corg and plushes workshop

* Fixes confinement cell access, fixes genpop interface access, adds nutrients to plushes workshop

* Adds old timer displays and windoors to solitary confinements, enables dynamic lighting for the workshop area, fixes bad interaction with the genpop locker

* lattice once again

* Fixes duplicated labor shuttle dock on Corg

* Wraps json_decode in a catch and try statement

* Swaps some rwalls for normal walls on corg, adds floorlights to workshop templates

* Fixes regex again, prevents secoff from locking unassigned prisoner lockers, minor map edits

* Update radstation

* Fixed Rad execution room being inaccessible

* Fixes bad integrity var and check for the turnstile, minor mapping tweaks.

Attempts to fix disposals in CorgStation

* Fixes hos windows instead of prison windows on Fland

* Comment on crime.dm indicating that it is overwritten by the config

* Various improvements (final maybe).

- Prisoners that are released will get pushed out by the turnstile
- The turnstile now removes PRISONER_ACCESS from IDs, fixed bad display time on prisoner IDs.
- Adds additional access airlock to the prison for security on all maps except Fland because no room on Fland
-Restricts access to SEC for the prison intake airlock.

* Bad leftover code smh head

* Fixes bad corgstation labor shuttle dock

* Bad extra windoor on a wall on Corg

* Improves prisoner lockers

Prisoner lockers can now be unassigned.

Improves their description.

* Batch of fixes and improvement

Fixes release/escape and time adjusting buttons not looking for any specific prisoner IDs.

Fixes bad time display for prison scanner.

Adds missing construction step for turnstile frames and adds rotation.

Properly implements the one directional turnstile with sprites. Replaces prison intake airlock with a turnstile.

Changes random lights in Boxstation I guess.

* Fixes bad cable on Corg

* Fixes make_plating being inconsistent in prison_floor.dm, fixes some bad lights

* Fixes bad disposals on Rad

* Hopefully removes all conflict markers

* Adresses requests

Adresses requests and fixes oversight on rad and meta.

Small refactor of CanAllowThrough() in genpop.dm. Buckled mobs and ridden vehicles are properly handled now.

Added check to prevent exit_push from executing if the prisoner left the turnstile turf.

* Bad airlock ruining my life fr

* Fixes inversed windows in warden office on delta

* Armsky floating in space on delta

* Update genpop.dm

* Fixes bad cables and zoning and pipes and wall navbeacon

Also makes south west solar on kilo functional again lol

* Rogue turnstile on fland

---------

Co-authored-by: PowerfulBacon <26465327+PowerfulBacon@users.noreply.github.com>
Co-authored-by: TsunamiAnt <TheTsunamiAnt@gmail.com>

ss13-beebot committed:
Automatic changelog generation for PR #10166 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

DrDuckedGoose committed:
Floor Resprite - The Sequel [MDB IGNORE] (#10276)

* basic - 1 10 23

* decal - 2 10 23

* Update decals.dmi

* Update floors.dmi

* showroom & bathroom - 2 10 23

* chapel - 2 10 23

* cult - 3 10 23

* Update floors.dmi

* non slip - 4 10 23

* inital - 10 10 23

* e

* grass - 11 10 23

* maints - 11 10 23

* better blending - 11 10 23

* better blending - 11 10 23

* Better settings & textures - 12 10 23

* Update areas.dm

* Procedural damage initial - 12 10 23

* More implementation & helpers - 12 10 23

* Blocking implementation - 12 10 23

* Update turf_damage.dmi

* better icon - 13 10 23

* better icon, for real - 13 10 23

* plaque - 15 10 23

* Update floors.dmi

* Update floors.dmi

* Floor variance - 15 10 23

* Better turf textures -  15 10 23

* Update Space_Station_13_areas.dm

* wood - 16 10 23

* grass - 16 10 23

* decals 1 - 17 10 23

* Decals 2- 17 10 23

* mapping - 18 10 23

* Update decals.dmi

* Update decals.dmi

* Update decals.dmi

* decals again - 20 10 23

* Update decals.dmi

* mo wood - 22 10 23

* bookmark - 25 11 23

* bookmark - 25 11 23

* better broken stuff - 4 12 23

* even better - 4 12 23

* e

* Grime texture tweaks - 5 12 23

* D:

* Swap overlays to filters for goofs - 6 12 23

* bookmark - 6 12 23

* Better floor damage icons - 6 12 23

* oops

* better blacklisting - 6 12 23

* Cleaning - 6 12 23

* Reorganize - 7 12 23

* Remapping - 7 12 23

* bruh

* Organize - 7 12 23

* Fix cleaning - 7 12 23

* Update turf_texture.dm

* Update turf_texture.dm

* Update turf.dm

* Mapping - 7 12 23

* Update MetaStation.dmm

* Grimy - 9 12 23

* Floor variance - 9 12 23

* Decals - 9 12 23

* Update beestation.dme

* Improve damage states - 10 12 23

* Steal grass from Paradise - 13 12 23

* Better turf offsets - 14 12 23

* Grass - 16 12 23

* Port that stuff - 16 12 23

* fugg

* review - 20 12 23

* More opt - 20 12 23

* oops

* Fix broken states - 20 12 23

* oops - 20 12 23

* update monotiles - 20 12 23

* Update turf_texture.dm

* Fix noslip - 22 12 23

* More globals - 22 12 23

* Update decals.dmi

* Update _open.dm

* rename procs - 7 1 24

* Pigeon Requests - 9 1 24

* Update misc_plating.dm

* Requests - 11 1 24

* noslip - 11 1 24

* Update fancy_floor.dm

* Update turf.dm

* Review - 11 1 124

* Update fancy_floor.dm

* Generic materials - 15 1 24

* Update MetaStation.dmm

* Update MetaStation.dmm

* Update a tile icon - 15 1 24

* Update mapping_helpers.dmi

* Update mapping_helpers.dmi

ss13-beebot committed:
Automatic changelog generation for PR #10276 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

DrDuckedGoose committed:
Fix Mapping & Icon Issues Related to Genpop (#10484)

* initial

* Revert "initial"

This reverts commit fa87a2a1d3938f8bef201b7b22b650e790612c8f.

* initial

* Update turfs.dm

ss13-beebot committed:
Automatic changelog generation for PR #10484 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Absolucy committed:
Fix ghosts being able to open/close cryotubes (#10476)

ss13-beebot committed:
Automatic changelog generation for PR #10476 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

itsmeow committed:
NewFood UpdatePaths Fix [MDB IGNORE] (#10493)

* Fix UpdatePaths

* More path fixes

* Reset ALL maps to old versions.

* Add raw types

* Fix knives UpdatePaths

* Add emitter UpdatePaths

* Add vendor and inventory adjustments repaths

* Mooore updatepaths

* Repath all maps with new scripts

* Apply inventory adjustment updepaths

* Add @SKIP to state in emitter

* Reset maps to try again

* Add cockroach repath

* Add monkeycubes

* Apply new UpdatePaths scripts

* Fix emitter script

* Reset maps again

* Apply UpdatePaths to maps

* It's really not my fault that UpdatePaths is busted. This one should show it working

* Fix all the emitters

* This one should work properly.

* Cockroach paths busted too. Can't fix it

* This should fix @SKIP

* Reset maps again again!!

* Fix @SKIP again

* Fix UpdatePaths @SKIP once and for all!

* Apply the UpdatePaths. This one will work I know it

* Add more sandwiches

* Fix varedited vending machine. UpdatePaths can't get that.

* UpdatePaths with sandwiches

* NEWFOOOD GRAAAAAAAAAAAAH

* PLEASE WORK IM BEGGING YOU!

* Fix vendor extended_inventory

* Fix jellied toast

* Reset any paths that aren't scoped in this PR

* Remove nugget subtypes

* Reset more paths.

* All the maps work! Reset to master now.

* Fix kitchen knife

ss13-beebot committed:
Automatic changelog generation for PR #10493 [ci skip]

Haliris committed:
Cleans up turnstile code and fixes it properly. (#10494)

Turnstiles now work properly again.

Simplified the code and axed a lot of potential interactions for the sake of sanity.

Co-authored-by: PowerfulBacon <26465327+PowerfulBacon@users.noreply.github.com>

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog generation for PR #10494 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Miliviu committed:
Fixes force feeding (#10495)

ss13-beebot committed:
Automatic changelog generation for PR #10495 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

itsmeow committed:
Update CODEOWNERS (#10501)

* Update CODEOWNERS

* Remove Corg from bacon on request

ss13-beebot committed:
Automatic changelog compile [ci skip]

Nakalon committed:
BeeShop Bug Fixes - White Hoodie and Dope necklace (#10363)

* fixes the bug

farts

* double dope necklace fix

* Changed the display name back

ss13-beebot committed:
Automatic changelog generation for PR #10363 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

EvilDragonfiend committed:
Adjusts Sanity Disruptor anomaly, minor nerf to fake wiz robe (#10486)

* anom fix

* anomaly fix

* revert stun time

* robe

* revert knuck

ss13-beebot committed:
Automatic changelog generation for PR #10486 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Update TGS DMAPI (#10482)

ss13-beebot committed:
Automatic changelog compile [ci skip]

itsmeow committed:
Fix ByondUI zombies causing lag (#10512)

ss13-beebot committed:
Automatic changelog generation for PR #10512 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ToasterBan committed:
Raw Khinkali can be cooked again (#10508)

PestoVerde322 committed:
GenPop workshop alignment and air disto removal (#10502)

* Align and remove the distros from the workshop holodecks

* eeee

---------

Co-authored-by: PigeonVerde322 <pesto.verde322@gmai.com>

ss13-beebot committed:
Automatic changelog generation for PR #10508 [ci skip]

ss13-beebot committed:
Automatic changelog generation for PR #10502 [ci skip]

ToasterBan committed:
Portable Seed Extractor now has an action button (#10497)

* Seed Extractor now has an action button

* removes newline

ss13-beebot committed:
Automatic changelog generation for PR #10497 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Penwin0 committed:
fixes saline drip icon (#10510)

Absolucy committed:
Fix ghosts being able to interact with the holoparasite HUD (#10525)

ss13-beebot committed:
Automatic changelog generation for PR #10510 [ci skip]

ss13-beebot committed:
Automatic changelog generation for PR #10525 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Nakalon committed:
Winter Coat Addition - Mail Bag Tweak (#10507)

* Adds thiings

* i had shit bitch broken

* final

* i forgot this

* uggy

* adresses Ruko's concerns

---------

Co-authored-by: RKz <falldownftw@gmail.com>

ss13-beebot committed:
Automatic changelog generation for PR #10507 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Absolucy committed:
Fix heretics getting themselves as a sacrifice target (#10524)

Penwin0 committed:
fixes heretic head sac not completing (#10523)

ss13-beebot committed:
Automatic changelog generation for PR #10523 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

XeonMations committed:
Ushanka sprite fix (#10522)

* ushanka

* this *should* fix the conflict?

i dont actually know what im doing

* Bump

* new sprites

ss13-beebot committed:
Automatic changelog generation for PR #10522 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

DrDuckedGoose committed:
Fix Floor Changes (#10488)

* initial

* Revert "initial"

This reverts commit fa87a2a1d3938f8bef201b7b22b650e790612c8f.

* initial - 16 1 24

* Fixes - 17 1 24

* Script - 18 1 24

* Update 10488_FLOOR_DAMAGE.txt

ss13-beebot committed:
Automatic changelog generation for PR #10488 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Nakalon committed:
Washing Machine - Coolification Expansion (#10371)

* Update 1 - Qdels and Sneaker Fix

the Colour Source no longer qdels
fixes coloring sneakers to actually work now

* Adds Security Dye, New Purple Suit

also adds Dye Colour to Sec SOP and Space Law

* Sounds, Sec Dye and misc

Adds sounds, Sec Dye and makes all UNDER clothes DYABLE

* Cleaning and pets

Cloning fix, and remove unused code

* I missed this

unused proc

* All Jumpskirts Dyable

Finally they're all dyable, also including female looking clothes that are not exactly skirts but should count as such

* I missed this, also all gloves

AGAIN.

Expand Jumpskirts, change QM dye to Turtleneck

All gloves dyable

* forgot this also, checks are scary

* Add Caps to list of dyable items

changed CE dye a bit for gloves also

and Made Dyed Caps unflipable

* oups i broke cleaning

* QOL cleaning and Alt Click

* Banana and Practice Gun Clown Foolery

Adds practice gun and banana to dyables

* Banana peel update

* SOP book change

Adds boring dye colours for the SOP books that were missing and adds a qdel for SOP books being used as a dye

* adds berets

Adds berets to the dye system.

Also adds a white beret for the purposes of dying

Still needs some revision, mainly clown and Mime

* Big Ass changes

This shit too big I dont even remember what I did

* all shoes dyable

* rogue lines because ocd

* what was this doign here

* oh yeah i remember now, yeah i need this

* missed adding this

* solved redundancy

* fixes somethings

* adressing concerns

* missed this

description updates

* hope this passes the checks now

* fix

* adressed Dakae's concerns

* Ike's Else If

Co-authored-by: ike709 <ike709@users.noreply.github.com>

---------

Co-authored-by: RKz <falldownftw@gmail.com>
Co-authored-by: ike709 <ike709@users.noreply.github.com>

ss13-beebot committed:
Automatic changelog generation for PR #10371 [ci skip]

Nakalon committed:
Solene Forgot the License.txt (#10538)

* license

* okay final

ss13-beebot committed:
Automatic changelog generation for PR #10538 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Hardly3D committed:
Adds unused clothing to drobes and brig phys winter coat to loadout (#10478)

* Adds stuff

* Update loadout_suit.dm

XeonMations committed:
Update catwalk.dm (#10516)

ss13-beebot committed:
Automatic changelog generation for PR #10478 [ci skip]

ss13-beebot committed:
Automatic changelog generation for PR #10516 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Miliviu committed:
Do not check if there is a door, check if there is a airlock or windoor (#10460)

ss13-beebot committed:
Automatic changelog generation for PR #10460 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

MarkusLarsson421 committed:
Changed the Labour Shuttle and Added Recharger (#10479)

* Changed the Labour shuttle and added recharger.

* GPS sec skin

* wip

* forgot to push oops

* Updated labour shuttles after feedback

---------

Co-authored-by: Nakalon <94647521+Nakalon@users.noreply.github.com>

ss13-beebot committed:
Automatic changelog generation for PR #10479 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

XeonMations committed:
properly fixed this time (#10541)

ss13-beebot committed:
Automatic changelog generation for PR #10541 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

MarkusLarsson421 committed:
Update brig_physician.dm (#10545)

ss13-beebot committed:
Automatic changelog generation for PR #10545 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

JixS4v committed:
Lowers zombie infection chance when hit (#10544)

* infection chance

* Ratón fucked up

* Forgive my stupidity

* Document.

* Auto something idk

ss13-beebot committed:
Automatic changelog generation for PR #10544 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

itsmeow committed:
Rust-g IconForge: Lightning fast spritesheet generation (#10404)

* Initial testing

* Rustg icon generator

* this thing.. fast....

* Fix

* Finish porting spritesheets and add GAGS support

* Remove 'moving' attribute

* Refactor the DM component

* Update rustg DM and add cleanup call

* vendor_icon_preview -> icon_state_preview

* Revert tracy change

* Re-add GAGS to design spritesheet

* Fix crafting spritesheet size

* Correct webroot args

* Adds caching

* Reset rustg

* Move cleanup step where it belongs

* Add crosscompiled rustg dll

* Log cache invalidations

* Fix player panel and orbit spritesheets

* Fix reading legacy sheets from cache

* Enables local caching of legacy assets, adds Regenerate Asset Cache debug verb

* Fix universal_icon creation being slow

* Protect against cache deletion

* Fix hand icons not displaying properly

* Remove this very incorrect assumption

* Restore rust_g to master

* Update defs

* Add rustg version checker to smart cache

* Move legacy cache to data/spritesheets so it persists on prod

* Update config comment to match

* Copy all icons so CI works

* Try fix CI again

* Fix chat and designs not displaying properly when loaded from cache

* Fix mistake

* Fix double encoding

* Inline uni_icon, saving unnecessary proc-call overhead

* Create a unit test for the smart cache

ss13-beebot committed:
Automatic changelog generation for PR #10404 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

MarkusLarsson421 committed:
Fixed Detectives Office and AI Upload Connection (#10553)

* Fixed detectives office and AI upload connection

* Update BoxStation.dmm

ss13-beebot committed:
Automatic changelog generation for PR #10553 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Rukofamicom committed:
Left-click unequips trashbags now (#10469)

Tsar-Salat committed:
Converts blob overmind shield to update_appearance (#10450)

* clean er up

* Update code/modules/antagonists/blob/structures/shield.dm

EvilDragonfiend committed:
Optimises ISEVEN() , ISODD() macro (#10504)

* optim even odd

* very mild optimisation

* == 0 is unnecessary

* Update code/__DEFINES/maths.dm

Co-authored-by: itsmeow <itsmeow@itsmeow.dev>

---------

Co-authored-by: itsmeow <itsmeow@itsmeow.dev>

ss13-beebot committed:
Automatic changelog generation for PR #10469 [ci skip]

ss13-beebot committed:
Automatic changelog generation for PR #10504 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Hardly3D committed:
Update language.dm (#10558)

ss13-beebot committed:
Automatic changelog generation for PR #10558 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

BriggsIDP committed:
Adds prefabs (#10557)

* Adds prefabs

* wall mounts too

ss13-beebot committed:
Automatic changelog generation for PR #10557 [ci skip]

ClownMoff committed:
KiloStation Medbay Rework 4 PR and FINAL (#10483)

* Update KiloStation.dmm

* Update KiloStation.dmm

* Update KiloStation.dmm

* Update KiloStation.dmm

* Update KiloStation.dmm

* Update KiloStation.dmm

* Update KiloStation.dmm

* MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE MERGE

* Update KiloStation.dmm

ss13-beebot committed:
Automatic changelog generation for PR #10483 [ci skip]

BriggsIDP committed:
Fland genpop update (#10550)

* working in rad because fland is huge

* should probably move this over

* map shuffling

* more mapping

* now i sleep

* final ammends

* Update FlandStation.dmm

* done?

* final

* update

* pigeon and markus comments

* woopsie

* spare lockers in maints

* Update FlandStation.dmm

* pigeon requested changes and ambiic update

ss13-beebot committed:
Automatic changelog generation for PR #10550 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

XeonMations committed:
Fixed Kilostation security. (#10543)

* kilostation security

To do: Wiring

* Wiring done.

* fix linter

* Update KiloStation.dmm

Camera ranges checked and fixed, removed windows in evidence, fixed access requirements on evidence.

* Update KiloStation.dmm

* DO NOT USE THIS ONE, ITS FOR AMBIIC

* do not use

* Update KiloStation.dmm

* for ambiic, a third time

* table fix

* Update KiloStation.dmm

* Update KiloStation.dmm

* Update KiloStation.dmm

* FOR YOU AMBIIC

* For you Xeon <3

* fixed merge error

* Update KiloStation.dmm

* Update KiloStation.dmm

* push

* Update KiloStation.dmm

* Update KiloStation.dmm

* Update KiloStation.dmm

* LET THERE BE LIGHT

* Update KiloStation.dmm

* Update KiloStation.dmm

* Update KiloStation.dmm

* Update KiloStation.dmm

* Update KiloStation.dmm

* LIGHT SWITCH

* CONFLICT FIX

* fix linters pls

* Update KiloStation.dmm

* Update KiloStation.dmm

* Update KiloStation.dmm

* Fixed areas

* Update KiloStation.dmm

* Moved rad shelter APC

* Moved rad shelter APC into the rad shelter

* Rerouted the Virology pipe.

* Update KiloStation.dmm

* Update KiloStation.dmm

* Update KiloStation.dmm

* Update KiloStation.dmm

* Update KiloStation.dmm

* Update KiloStation.dmm

* Update KiloStation.dmm

---------

Co-authored-by: Markus <markus.larsson421@gmail.com>

ss13-beebot committed:
Automatic changelog generation for PR #10543 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Dejaku51 committed:
im not sure what im going

Dejaku51 committed:
Merge branch 'servers' of https://github.com/Dejaku51/BeeStation-Hornet into servers
